core/controllers/acl_decorators.py

```diff
@@ -2615,3 +2617,140 @@ def test_can_play_entity(self, entity_type, entity_id, **kwargs):
     test_can_play_entity.__wrapped__ = True
 
     return test_can_play_entity
+
+
+def can_view_submitted_voiceover_applications(handler):
+    """Decorator to check whether the user can view submitted voiceover
+    applications.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if a
+            user has permission to view submitted voiceover applications.
+    """
+
+    def test_can_view_submitted_voiceover_applications(self, **kwargs):
+        """Checks if the user can view submitted voiceover applications.
+
+        Returns:
+            *. The return value of the decorated function.
+
+        Raises:
+            NotLoggedInException: The user is not logged in.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        return handler(self, **kwargs)
+
+    test_can_view_submitted_voiceover_applications.__wrapped__ = True
+
+    return test_can_view_submitted_voiceover_applications
+
+
+def can_review_voiceover_application(handler):
+    """Decorator to check whether the user can review voiceover applications.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that now also checks if a
+            user has permission to review voiceover application.
+    """
+
+    def test_can_review_voiceover_application(
```

Comment:
Seems a bit broad. Shouldn't it be language-proficiency-specific?

If deferring that part to later, please add a TODO (and an issue).

Reply:
> Shouldn't it be language-proficiency-specific?

Currently, we don't have `proficiency-structure` (not even a term in our codebase) once we have the `proficiency-structure` in the codebase we are going to change the decorator based on the requirement.

I think this decorator is working as expected (Allowing admin to review the application) as of now so I'm not sure whether we should add TODO? (We do have an issue for the proficiency structure project at https://github.com/oppia/oppia/issues/6671)

Reply:
Ah, I see. But it looks like this won't be a sustainable long-term solution, so yes, please go ahead and add a TODO.